File: backend/insurance-management/src/main/java/com/aurionpro/app/service/AgentService.java
package com.aurionpro.app.service;

import java.util.List;

import com.aurionpro.app.dto.AgentAssignedPolicyDTO;
import com.aurionpro.app.dto.AgentProfileDTO;
import com.aurionpro.app.dto.AgentResponseDTO;
import com.aurionpro.app.dto.PageResponse;

public interface AgentService {
	PageResponse<AgentResponseDTO> getPendingAgents(int pageNumber, int pageSize);
    AgentResponseDTO approveAgent(int agentId, int approverId);
    void softDeleteAgent(int id);
    List<AgentAssignedPolicyDTO> getAssignedPolicies(int agentId);
    AgentProfileDTO getProfile(int agentId);
    AgentResponseDTO rejectAgent(int agentId, int approverId);
}

File: frontend/insurance-mgmt/src/app/employee/components/services/agent.service.ts
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { AgentResponseDTO } from '../../../models/agent.model';
import { PageResponse } from '../../../models/page.model';

@Injectable({
  providedIn: 'root'
})
export class AgentService {
  private baseUrl = 'http://localhost:8080/app'
  private apiUrl = `http://localhost:8080/app/agent`

  constructor(private http: HttpClient) { }

  registerAgent(agentData: any): Observable<any> {
    const token = localStorage.getItem('accessToken');
    const headers = new HttpHeaders({
      Authorization: `Bearer ${token}`,
    });
  
    return this.http.post(`${this.baseUrl}/register/agent`, agentData, { headers });
  }
  
  getPendingAgents(page: number, size: number): Observable<PageResponse<AgentResponseDTO>> {
    const token = localStorage.getItem('accessToken');
    const headers = new HttpHeaders({
      Authorization: `Bearer ${token}`,
    });
  
    return this.http.get<PageResponse<AgentResponseDTO>>(
      `${this.apiUrl}/pending?page=${page}&size=${size}`,
      { headers }
    );
  }  

  approveAgent(agentId: number): Observable<any> {
    const approverId = localStorage.getItem('userId');
    return this.http.put(`${this.apiUrl}/approve/${agentId}?approverId=${approverId}`, {});
  }

  rejectAgent(agentId: number): Observable<any> {
    const approverId = localStorage.getItem('userId');
    return this.http.put(`${this.apiUrl}/reject/${agentId}?approverId=${approverId}`, {});
  }
}
